content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md

```diff
@@ -0,0 +1,488 @@
+instead of a `serviceMonitorSelector`.
+
+{{% /alert %}}
+
+### Did you leave out the serviceMonitorSelector and/or podMonitorSelector configuration altogether?
+
+As mentioned in
+["Did you configure a ServiceMonitor or PodMonitor selector"](#did-you-configure-a-servicemonitor-or-podmonitor-selector),
+setting mismatched values for `serviceMonitorSelector` and `podMonitorSelector`
+results in the Target Allocator failing to discover scrape targets from your
+`ServiceMonitors` and `PodMonitors`, respectively.
+
+Similarly, in
+[`v1beta1`](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollector-1)
+of the `OpenTelemetryCollector` CR, leaving out this configuration altogether
+also results in the Target Allocator failing to discover scrape targets from
+your `ServiceMonitors` and `PodMonitors`.
+
+As of `v1beta1` of the `OpenTelemetryOperator`, a `serviceMonitorSelector` and
+`podMonitorSelector` must be included, even if you don’t intend to use it, like
+this:
+
+```yaml
+prometheusCR:
+  enabled: true
+  podMonitorSelector: {}
+  serviceMonitorSelector: {}
+```
+
+See the
+[full OpenTelemetryCollector definition in "Do you know if metrics are actually being scraped?"](#do-you-know-if-metrics-are-actually-beingscraped).
+
+### Do your labels, namespaces, and ports match for your ServiceMonitor and your Service (or PodMonitor and your Pod)?
+
+The `ServiceMonitor` is configured to pick up Kubernetes
+[Services](https://kubernetes.io/docs/concepts/services-networking/service/)
+that match on:
+
+- Labels
+- Namespaces (optional)
+- Ports (endpoints)
+
+Suppose that you have this `ServiceMonitor`:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: sm-example
+  labels:
+    app: my-app
+    release: prometheus
+spec:
+  selector:
+    matchLabels:
+      app: my-app
+  namespaceSelector:
+    matchNames:
+      - opentelemetry
+  endpoints:
+    - port: prom
+      path: /metrics
+    - port: py-client-port
+      interval: 15s
+    - port: py-server-port
+```
+
+The previous `ServiceMonitor` is looking for any services that have:
+
+- the label `app: my-app`
+- reside in a namespace called `opentelemetry`
+- a port named `prom`, `py-client-port`, _or_ `py-server-port`
+
+For example, the following `Service` resource would get picked up by the
+`ServiceMonitor`, because it matches the previous criteria:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: py-prometheus-app
+  namespace: opentelemetry
+  labels:
+    app: my-app
+    app.kubernetes.io/name: py-prometheus-app
+spec:
+  selector:
+    app: my-app
+    app.kubernetes.io/name: py-prometheus-app
+  ports:
+    - name: prom
+      port: 8080
+```
+
+The following `Service` resource would not be picked up, because the
+`ServiceMonitor` is looking for ports named `prom`, `py-client-port`, _or_
+`py-server-port`, and this service’s port is called `bleh`.
```

Comment:
This is such a great example and _exactly_ what I helped a user debug in slack a few weeks ago! Thanks for including it.

Reply:
❤️